commcare_connect/form_receiver/processor.py

```diff
@@ -206,7 +209,12 @@ def process_deliver_unit(user, xform: XForm, app: CommCareApp, opportunity: Oppo
     if flags:
         user_visit.flagged = True
         user_visit.flag_reason = {"flags": flags}
+
+    if opportunity.auto_approve_visits and user_visit.status == VisitValidationStatus.pending:
+        user_visit.status = VisitValidationStatus.rejected if user_visit.flagged else VisitValidationStatus.approved
     user_visit.save()
+    if opportunity.auto_approve_payments:
+        approve_completed_work_and_update_payment_accrued.delay([completed_work.id])
```

Comment:
I would be a bit concerned about sequencing here. It may not matter but forms can be submitted many at once (since they will submit in bulk once the user gets data). We may want locking on this task, or to have it run on a schedule in bulk instead of immediately (which may make more sense anyway to do it as a batch)

Reply:
The bulk processing for the payment approval makes sense. I will change the code to run this task periodically every day using celery schedules.